### PR TITLE
add validation for database replication factor and write concern

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Validate replicationFactor and writeConcern when creating databases

- [ ] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Bug-Fix for a *released version* (did you remember to port this to all relevant release branches?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *scripts/unittest shell_server --test one-shard --cluster true*.

https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7602/

### CODE REVIEW

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7607/

File: tests/js/server/shell/shell-general-graph-creation-cluster.js [hunk 1]
```diff
@@ -56,6 +56,8 @@ function GeneralGraphClusterCreationSuite() {
         graph._drop(gn, true);
       } catch (ignore) {
       }
+      db._drop(vn);
```
Comment (hkernbach) [inline tests/js/server/shell/shell-general-graph-creation-cluster.js]: Why is that necessary? `vn` and `en` should be dropped automatically with `graph._drop(gn, true);`

**Comment by jsteemann** [comment]
https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7610/

File: tests/js/server/shell/shell-general-graph-creation-cluster.js (hunk 1 above)
Comment (jsteemann) [reply tests/js/server/shell/shell-general-graph-creation-cluster.js]: I got a failure once that the two collections (vertex & edge) were not cleaned up after the test. So I added the drop commands for extra safety.
Shouldn't do any harm

**Comment by jsteemann** [comment]
Tests blue
